plugins/parsers/value/parser.go

```diff
@@ -23,6 +23,7 @@ func (v *ValueParser) Parse(buf []byte) ([]telegraf.Metric, error) {
 	if v.DataType == "string" {
 		vStr = strings.TrimSpace(string(buf))
 	} else {
+		buf = bytes.Trim(buf, "\x00")
```

Comment:
does bytes.TrimSpace trim null characters? maybe we should just do `buf = bytes.TrimSpace(buf)` as the first step in this function, then we can also remove the strings.TrimSpace if the data type is a string as well.